src/Features/Core/Portable/ExtractMethod/ExtractMethodMatrix.cs

```diff
@@ -114,6 +114,8 @@ private static void BuildMatrix()
             s_matrix.Add(new Key(dataFlowIn: false, dataFlowOut: false, alwaysAssigned: true, variableDeclared: false, readInside: false, writtenInside: true, readOutside: false, writtenOutside: true), VariableStyle.SplitIn);
             s_matrix.Add(new Key(dataFlowIn: false, dataFlowOut: false, alwaysAssigned: true, variableDeclared: false, readInside: false, writtenInside: true, readOutside: true, writtenOutside: false), VariableStyle.SplitIn);
             s_matrix.Add(new Key(dataFlowIn: false, dataFlowOut: false, alwaysAssigned: true, variableDeclared: false, readInside: false, writtenInside: true, readOutside: true, writtenOutside: true), VariableStyle.SplitIn);
+            s_matrix.Add(new Key(dataFlowIn: false, dataFlowOut: false, alwaysAssigned: true, variableDeclared: false, readInside: true, writtenInside: false, readOutside: false, writtenOutside: true), VariableStyle.InputOnly);
```

Comment:
dataflowin, dataflowout being false, and readinside true and variabledeclared false is not something expected for a symbol. if it is not declared inside and read, data must flow in. but not for unreachable code case.

Reply:
here we have 8 variables, which give us 256 possibilities. among those, these are expected combination. we don't expect combination that not belong to ones listed here rather than silently ignore those case or create random code or show some dialog box to users.

I agree we should not crash VS. and we don't.